Dockerfile

```diff
@@ -27,8 +27,10 @@ RUN --mount=type=cache,target=/var/cache/apt \
     # Configure RStudio Server to run without auth
     echo "auth-none=1" >> /etc/rstudio/rserver.conf &&\
     echo "USER=rstudio" >> /etc/environment &&\
-    # Remove the packages shipped with the rocker image
-    rm -rf /usr/library/lib/R/site-library/* &&\
+    # Remove the packages shipped with the rocker image (apart from required base packages)
+    find /usr/local/lib/R/library/ -mindepth 1 -maxdepth 1 -type d |\
+    grep -v -E 'base|codetools|compiler|graphics|grDevices|grid|methods|parallel|splines|stats|stats4|tcltk|tools|translations|utils' |\
+    xargs -I{} rm -rf {} &&\
```

Comment:
This looks brittle. Why do we need to call `find`? Does the directory move between releases? How do we know that these packages are the _only_ required base packages? Could you link to some documentation?

Reply:
I agree it feels a bit brittle. The tradeoff is we leave the packages intact and someone references them in an R script which then won't work in the OpenSAFELY R docker container. `find` is merely a tool to enumerate the directory contents here, I decided that a strategy of "keep these" is better than "delete these" in case of a new package being added upstream by Rocker.

This list was derived by filtering the installed packages with "base" priority

```R
installed_packages <- as.data.frame(installed.packages()[,c(1,3:4)])
package_names<-unique(installed_packages[installed_packages$Priority=="base",,drop=FALSE][["Package"]])
package_names
[1] "base"      "compiler"  "datasets"  "graphics"  "grDevices" "grid"      "methods"   "parallel"  "splines"   "stats"     "stats4"    "tcltk"     "tools"    
[15] "utils"  
```

Reply:
plus `codetools` to support code completion, and `translations` in case we have any users who have a non-English locale set

Reply:
Thinking about this some more, rather than deletion of the package dirs, we could uninstall them from within R using this list derivation (which itself runs in R) as part of the post-setup script. The preference has been to have as much as possible in the Dockerfile rather than the devcontainer config, but this might be a better fit in the latter

Reply:
However, doing this after we've dropped in the renv packages would be a right pain as we'd have to filter those out too. Not impossible, just a pain.

Reply:
So, if I've understood you correctly, then:

* RStudio comes with several pre-installed packages
* We want to remove them, such that a user can't import them from a script that executes in a dev container but doesn't execute in an R action container
* We thought they lived in `/usr/library/lib/R/site-library`
* They live in `/usr/local/lib/R/library`
* We don't want to remove them all, because several make RStudio work correctly
* We hard-code a list of the packages we want to keep
* We remove the packages that aren't on the list

How close am I?

Reply:
spot on

Reply:
Could we simply not remove the pre-installed packages and accept the risk?

Reply:
I'm inclined to agree. If it becomes a regular problem then we can address it in future

Reply:
This was also motivated by making the "are all the R packages" test easier - i.e. so that the count of (base + user) packages == count of packages copied in from renv

Reply:
We could apply the same filtering that I proposed here in that test instead, I guess. If new packages are added then the test will fail. This feels like a good thing?

Reply:
Where is that test?

Reply:
Under discussion in @StevenMaude 's issue for creating tests! Purely hypothetical right now, absolutely not a reason  to preserve this PR

Reply:
Okay, so, if...

```sh
rm -rf /usr/library/lib/R/site-library/* &&\
```

isn't doing anything because that directory doesn't exist, then this PR should remove it.

Reply:
Correct, good point. I'll alter the solitary commit to that effect and force push